### PR TITLE
[Feature] Support REPLACE TABLE operation.

### LINKED ISSUE #4668: [Proposal] Support replace table feature
**Is your feature request related to a problem? Please describe.**
in #2663, I implemented a feature of `Temporary Partition`, so that users can use this function to
atomically write data to a partition. But this function is only for partition tables.

Therefore, here I have implemented a replacement table operation, which can perform atomic replacement of the table,
so that for non-partitioned tables, this method can also be used to achieve the purpose of atomic overwriting.

Syntax:

```
ALTER TABLE tbl1 REPLACE WITH TABLE tbl2;
```

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

Implement `REPLACE TABLE` feature.
Details can be found in document `alter-table-replace-table.md`

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have create an issue on (Fix #4668), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document

### CODE REVIEW

File: docs/zh-CN/administrator-guide/alter-table/alter-table-replace-table.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+```
+ALTER TABLE [db.]tbl1 REPLACE WITH tbl2
+[PROPERTIES('swap' = 'true')];
+```
+
+将表 tbl1 替换为表 tbl2。
+
+如果 `swap` 参数为 `true`，则替换后，名称为 `tbl1` 表中的数据为原 `tbl2` 表中的数据。而名称为 `tbl2` 表中的数据为原 `tbl1` 表中的数据。即两张表数据发生了互换。
+
+如果 `swap` 参数为 `true`，则替换后，名称为 `tbl1` 表中的数据为原 `tbl2` 表中的数据。而名称为 `tbl2` 表被删除。
```
Comment (yangzhg) [inline docs/zh-CN/administrator-guide/alter-table/alter-table-replace-table.md:45]: should be false ？

File: docs/en/administrator-guide/alter-table/alter-table-replace-table.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+```
+ALTER TABLE [db.]tbl1 REPLACE WITH tbl2
+[PROPERTIES('swap' = 'true')];
+```
+
+Replace table `tbl1` with table `tbl2`.
+
+If the `swap` parameter is `true`, after replacement, the data in the table named `tbl1` is the data in the original `tbl2` table. The data in the table named `tbl2` is the data in the original table `tbl1`. That is, the data of the two tables are interchanged.
+
+If the `swap` parameter is `true`, after replacement, the data in the table named `tbl1` is the data in the original `tbl2` table. The table named `tbl2` is dropped.
```
Comment (yangzhg) [inline docs/en/administrator-guide/alter-table/alter-table-replace-table.md:45]: false ？

File: docs/zh-CN/administrator-guide/alter-table/alter-table-replace-table.md (hunk 1 above)
Comment (morningman) [reply docs/zh-CN/administrator-guide/alter-table/alter-table-replace-table.md:45]: fixed

File: docs/en/administrator-guide/alter-table/alter-table-replace-table.md (hunk 2 above)
Comment (morningman) [reply docs/en/administrator-guide/alter-table/alter-table-replace-table.md:45]: fixed

**Review by yangzhg — APPROVED** [review]
+1

**Pull request merged**